core/src/test/java/io/confluent/rest/Http2Test.java

```diff
@@ -0,0 +1,390 @@
+
+      int statusCode;
+      try {
+        statusCode = makeGetRequestHttp2(httpUri + "/test");
+        fail("HTTP/2 Cleartext should not be enabled");
+      } catch (java.util.concurrent.ExecutionException exc) {
+        // Fall back to HTTP/1.1 once we've seen HTTP/2C fail
+        assertTrue(exc.getCause() instanceof java.net.ConnectException);
+        statusCode = makeGetRequestHttp(httpUri + "/test");
+        assertEquals(EXPECTED_200_MSG, 200, statusCode);
+      }
+      assertMetricsCollected();
+    } finally {
+      app.stop();
+    }
+  }
+
+  @Test
+  public void testHttp2NotEnabled() throws Exception {
+    TestMetricsReporter.reset();
+    Properties props = new Properties();
+    String httpsUri = "https://localhost:8081";
+    props.put(RestConfig.LISTENERS_CONFIG, httpsUri);
+    props.put(RestConfig.METRICS_REPORTER_CLASSES_CONFIG, "io.confluent.rest.TestMetricsReporter");
+    props.put(RestConfig.HTTP2_ENABLED_CONFIG, false);
+    configServerKeystore(props);
+    TestRestConfig config = new TestRestConfig(props);
+    Http2TestApplication app = new Http2TestApplication(config);
+    try {
+      app.start();
+
+      int statusCode;
+      try {
+        statusCode = makeGetRequestHttp2(httpsUri + "/test",
+                                         clientKeystore.getAbsolutePath(), SSL_PASSWORD, SSL_PASSWORD);
+        fail("HTTP/2 Cleartext should not be enabled");
+      } catch (java.util.concurrent.ExecutionException exc) {
+        // Fall back to HTTP/1.1 once we've seen HTTP/2 fail
+        assertTrue(exc.getCause() instanceof java.net.ConnectException);
+        statusCode = makeGetRequestHttp(httpsUri + "/test",
+                                        clientKeystore.getAbsolutePath(), SSL_PASSWORD, SSL_PASSWORD);
+        assertEquals(EXPECTED_200_MSG, 200, statusCode);
+      }
+      assertMetricsCollected();
+    } finally {
+      app.stop();
+    }
+  }
+
+  private void assertMetricsCollected() {
+    assertNotEquals(
+        "Expected to have metrics.",
+        0,
+        TestMetricsReporter.getMetricTimeseries().size());
+    for (KafkaMetric metric : TestMetricsReporter.getMetricTimeseries()) {
+      if (metric.metricName().name().equals("request-latency-max")) {
+        Object metricValue = metric.metricValue();
+        assertTrue(
+            "Request latency metrics should be measurable",
+            metricValue instanceof Double);
+        double latencyMaxValue = (double) metricValue;
+        assertNotEquals(
+            "Metrics should be collected (max latency shouldn't be 0)",
+            0.0,
+            latencyMaxValue);
+      }
+    }
+  }
+
+  // returns the http response status code.
+  private int makeGetRequestHttp(String url) throws Exception {
+    log.debug("Making GET using HTTP " + url);
+    HttpClient httpClient = new HttpClient();
+    httpClient.start();
+
+    int statusCode = httpClient.GET(url).getStatus();
+    httpClient.stop();
+    return statusCode;
+  }
```

Comment:
I think we should compact the 4 `makeXxxRequestYyy` methods, extracting the common parts into separate methods. All of them do:
```
httpClient.start();

int statusCode = httpClient.GET(url).getStatus();
httpClient.stop();
return statusCode;
```
They only differ in the way they set up the used client - it needs to be an HTTP/2 client in the case of HTTP/2, and it needs to be set up with an `SslContextFactory` using some pre-defined keystore config in the case of HTTPS. I'd say let's extract the `SslContextFactory` creation in one method, and then the client creation depending on whether HTTPS is being used and on whether HTTP/2 is being used.